creation/web_base/condor_startup.sh

```diff
@@ -750,19 +760,35 @@ EOF
         # glidein_disk empty is the same as auto, used in the slot_type definitions
         glidein_disk=$(gconfig_get GLIDEIN_DISK "$config_file")
         # set up the slots based on the slots_layout entry parameter
-	slots_layout=$(gconfig_get SLOTS_LAYOUT "$config_file")
+	    slots_layout=$(gconfig_get SLOTS_LAYOUT "$config_file")
         if [[ "X$slots_layout" = "Xpartitionable" ]]; then
             echo "NUM_SLOTS = 1" >> "$CONDOR_CONFIG"
             echo "SLOT_TYPE_1 = cpus=\$(GLIDEIN_CPUS) ${glidein_disk:+disk=${glidein_disk}}" >> "$CONDOR_CONFIG"
             echo "NUM_SLOTS_TYPE_1 = 1" >> "$CONDOR_CONFIG"
             echo "SLOT_TYPE_1_PARTITIONABLE = True" >> "$CONDOR_CONFIG"
             num_slots_for_shutdown_expr=1
+            #Blackhole calculation based on the parent startd stats (child slots will have the same stats).
+            echo "STARTD_LATCH_EXPRS = GLIDEIN_BLACKHOLE" >> "$CONDOR_CONFIG"
+            echo "BLACKHOLE_TRIGGERED_P = (RecentJobBusyTimeAvg < (1.0/\$(glidein_blackhole_rate))) && (RecentJobBusyTimeCount >= \$(glidein_blackhole_numjobs))" >> "$CONDOR_CONFIG"
+            echo "GLIDEIN_BLACKHOLE = \$(glidein_blackhole) || ifThenElse(isUndefined(\$(BLACKHOLE_TRIGGERED_P)), False, \$(BLACKHOLE_TRIGGERED_P))" >> "$CONDOR_CONFIG"
+            echo "STARTD_ATTRS = \$(STARTD_ATTRS), GLIDEIN_BLACKHOLE, glidein_blackhole_numjobs, glidein_blackhole_rate" >> "$CONDOR_CONFIG"
+            echo "START = (\$(START)) && (\$(GLIDEIN_BLACKHOLE) =?= False)" >> "$CONDOR_CONFIG"
```

Comment:
Aren't glidein_blackhole_numjobs, glidein_blackhole_rate and glidein_blackhole local shell variables ("$glidein_blackhole_rate", not in STARTD_ATTRS)?
If you want also condor variables/configuration macros, where do you initialize them?
What is the condor config produced here?
Optionally, use a heredoc (cat << EOF >> "$CONDOR_CONFIG")

Reply:
Clarification received: 
* `glidein_blackhole_numjobs`, `glidein_blackhole_rate` and `glidein_blackhole` are local shell variables.
* does not make sense to use the shell variables as condor variables (escaped with a `\`) since `glidein_blackhole_numjobs` and `glidein_blackhole_rate` are constant values and do not change throughout the lifetime of the glidein. All that is needed is to substitute the values of the shell variables in condor expressions that are defined (uppercase names)
* translate the series of echo statements to the heredoc syntax (just like how the logic for static slots layout has used)